src/publisher.ts

```diff
@@ -418,8 +401,200 @@ export class WebsocketCacheProgramAccountSubscriber {
 	}
 }
 
+class grpcCacheProgramAccountSubscriber extends WebsocketCacheProgramAccountSubscriber {
+	client: Client;
+	stream: ClientDuplexStream<SubscribeRequest, SubscribeUpdate>;
+
+	constructor(
+		endpoint: string,
+		token: string,
+		program: Program,
+		redisClient: RedisClient,
+		options: { filters: MemcmpFilter[]; commitment?: Commitment } = {
+			filters: [],
+		},
+		resubTimeoutMs?: number | undefined
+	) {
+		super(program, redisClient, options, resubTimeoutMs);
+		this.client = new Client(endpoint, token, {
+			'grpc.keepalive_time_ms': 10_000,
+			'grpc.keepalive_timeout_ms': 1_000,
+			'grpc.keepalive_permit_without_calls': 1,
+			'grpc.max_receive_message_length': 100 * 1024 * 1024,
+		});
+	}
+
+	async subscribe(): Promise<void> {
+		await this.decoder.init();
+
+		if (this.listenerId != null || this.isUnsubscribing) {
+			return;
+		}
+
+		let syncCount = 0;
+		const syncInterval = setInterval(async () => {
+			logger.info('Syncing on interval');
+			await this.sync();
+
+			if (syncCount % EXPIRY_MULTIPLIER === 0) {
+				logger.info('Checking sync on interval');
+				await this.checkSync();
+			}
+
+			syncCount++;
+		}, SYNC_INTERVAL);
+
+		this.syncInterval = syncInterval;
+
+		if (SYNC_ON_STARTUP === 'true') {
+			await this.sync();
+		}
+
+		this.stream = await this.client.subscribe();
+		const filters = this.options.filters.map((filter) => ({
+			memcmp: {
+				offset: filter.memcmp.offset.toString(),
+				bytes: bs58.decode(filter.memcmp.bytes),
+			},
+		}));
+		const request: SubscribeRequest = {
+			slots: {},
+			accounts: {
+				drift: {
+					account: [],
+					owner: [this.program.programId.toBase58()],
+					filters,
+				},
+			},
+			transactions: {},
+			blocks: {},
+			blocksMeta: {},
+			accountsDataSlice: [],
+			commitment: CommitmentLevel.CONFIRMED,
+			entry: {},
+			transactionsStatus: {},
+		};
+
+		this.stream.on('data', (chunk: SubscribeUpdate) => {
```

Comment:
might be prudent to also handle events other events like error, end, pause, close. since we know how triton has been..and either reconnect or something